Application/EdFi.Ods.Api/Middleware/OdsInstanceSelector.cs

```diff
@@ -45,7 +49,7 @@ public async Task<OdsInstanceConfiguration> GetOdsInstanceAsync(IReadOnlyDiction
             throw new ApiSecurityConfigurationException("The API client has not been associated with an ODS instance.");
         }
 
-        if (apiClientContext.OdsInstanceIds.Count == 1)
+        if (apiClientContext.OdsInstanceIds.Count == 1 && !_odsRoutesRootTemplateProvider.HasOdsContextRouteTemplate)
```

Comment:
I would suggest introducing a private `Lazy<bool>` field on this class and initializing it in the constructor using a value factory delegate initialization that uses the result of the `odsRouteRootTemplateProvider.GetOdsRouteTemplate()` method. Then this logic would just check the `Value` of that field (reusing the result for the _extremely_ high volume of executions of this code).